client/src/components/Map.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { Component, useRef, useState, useEffect } from 'react';
+import * as d3 from 'd3';
+import ZoomableSVG from './ZoomableSVG';
+
+type Prop = {
+    layoutImage: string
+}
+
+export default class Map extends Component<Prop>{
+
+    mapRef!: SVGSVGElement
+
+    private drawMap(){
+
+        const svgElement = d3.select(this.mapRef)
+        .append("image")
+        .attr("xlink:href", `/submaps/${this.props.layoutImage}.svg`)
+        .attr("width", 450)
+        .attr("height", 300)  
+
+    }
+
+
+    private drawNodesOnClick(){
+
+        const svg = d3.select(this.mapRef)
+        svg.on("click", function(event) {
+
+            console.log(d3.pointer(event))
+
+            svg.append("circle")
+            .attr("cx", d3.pointer(event)[0])
+            .attr("cy", d3.pointer(event)[1])
+            .attr("r", 1.5)
+            .attr("fill","#41C7F7")
+            .attr("stroke-opacity", 1)
+            .attr("stroke-width", 1.5)
+
+        })
+    }
+
+    private connectNodes( {x1, y1, x2, y2} ){
+
+        const svg = d3.select(this.mapRef)
+
+        svg.append("circle")
+        .attr("cx", x1)
+        .attr("cy", y1)
+        .attr("r", 1.5)
+        .attr("fill","#41C7F7")
+        .attr("stroke-opacity", 1)
+        .attr("stroke-width", 1.5)
+        .classed("animate-pulse", true)
+
+        svg.append("circle")
+        .attr("cx", x2)
+        .attr("cy", y2)
+        .attr("r", 1.5)
+        .attr("fill","#41C7F7")
+        .attr("stroke-opacity", 1)
+        .attr("stroke-width", 1.5)
+        .classed("animate-pulse", true)
+
+        svg.append("line")
+        .attr("x1", x1)
+        .attr("y1", y1)
+        .attr("x2", x2)
+        .attr("y2", y2)
+        .style("stroke", "#41C7F7")
+        .style("stroke-width", 0.5)
+        .attr("stroke-linecap", "round")
+        .attr("stroke-opacity", 0.6)
+
+    }
+
+
+    componentDidMount() {
+        
+        this.drawMap()
+        this.drawNodesOnClick()
```

Comment:
This onClick option will be useful, but should not always be enabled. Add a boolean flag for it to the props. The click should be disabled by default (if the prop is not passed to the component).